src/Microsoft.Performance.SDK.Runtime/Extensibility/BaseSourceProcessingSession.cs

```diff
@@ -116,17 +116,18 @@ public DataProcessingResult ProcessDataElement(T data, TContext context, Cancell
 
             var key = data.GetKey();
 
-            if (!this.activeCookerRegistry.ContainsKey(key) && !this.receiveAllDataCookers.Any())
+            if (!this.activeCookerRegistry.ContainsKey(key) && this.receiveAllDataCookers.Count == 0)
```

Comment:
why this change?

Reply:
Using .Any() requires an enumerator. Since Count is cached, we can access that faster.